src/H5Cdbg.c

```diff
@@ -415,12 +405,7 @@ H5C_stats(H5C_t *cache_ptr, const char *cache_name,
 
     FUNC_ENTER_NOAPI(FAIL)
 
-    HDassert(cache_ptr->magic == H5C__H5C_T_MAGIC);
-
-    /* This would normally be an assert, but we need to use an HGOTO_ERROR
-     * call to shut up the compiler.
-     */
-    if ((NULL == cache_ptr) || (cache_ptr->magic != H5C__H5C_T_MAGIC) || (NULL == cache_name))
+    if (NULL == cache_ptr || NULL == cache_name)
         HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, FAIL, "Bad cache_ptr or cache_name")
```

Comment:
Not a big deal, but we could use H5_ATTR_UNUSED to "shut up the compiler" now

Reply:
Yes, true, I should review these

Reply:
Probably out of scope for this PR. We can clean in a subsequent one.